### PR TITLE
Update default Exit This Page button text

### LINKED ISSUE #3688: The non-visual experience of Exit this Page button doesn't make it clear it's a safety feature
This issue is from May 2023 external accessibility audit report.

**Usability issue**

**Issue ID:**
DAC_Usability_Exit_Button_01

**URLs:**
[https://govuk-frontend-pr-3503.herokuapp.com/full-page-examples/child-maintenance](https://govuk-frontend-pr-3503.herokuapp.com/full-page-examples/child-maintenance)

## Screen shots
Exit this page red button

The ‘Exit this page’ button did not inform users of screen reading assistive technologies of its emergency purpose with the label. Although this may already be conveyed on a previous page, it does not mean that the user will always remember this information.

## Current code ref(s)
`#main-content > div.govuk-exit-this-page > a`

> However, although I was aware from the label of the button that I would exit the page it was not immediately clear to me that the item was an emergency exit and that my information would be removed from the page. Placing the phrase ‘Emergency’ within the label for example ‘Emergency exit from this page’ with additional hint information advising me that the page will be removed from browsing history would ensure that all screen reader users are aware of the value of this feature.”

## Solution
Consider adding the term ‘Emergency’ to the button. This can be visually hidden to provide that extra information for users of screen reading assistive technologies, while keeping the current aesthetics.

### PR DESCRIPTION
**Pull request by querkmachine**
Updates the default Exit This Page button text to include visually hidden text, so the button reads 'Emergency Exit this Page' to screen reader users, to clarify the button's intended use as a safety feature. 

Split out from #3878. Intended to resolve #3688. 

I assume this would need a 'Suggested change' in the changelog now that EtP has shipped in 4.7.0?

### CODE REVIEW

File: packages/govuk-frontend/src/govuk/components/exit-this-page/template.njk [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {% from "../button/macro.njk" import govukButton -%}
 
+{% set hasSetCustomText = params.html !== undefined or params.text !== undefined -%}
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/exit-this-page/template.njk:3]: There's a gotcha in this as `{ html: null }` will pass the checks

Probably too far to filter empty space with ` | trim` but will this do?

```suggestion
{% set hasSetCustomText = params.html or params.text -%}
```

File: packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.yaml [hunk 2]
```diff
@@ -2,11 +2,11 @@ params:
   - name: text
     type: string
     required: false
-    description: Text for the link. If `html` is provided, the `text` option will be ignored. Defaults to 'Exit this page'.
+    description: Text for the link. If `html` is provided, the `text` option will be ignored. Defaults to 'Emergency Exit this page' with 'Emergency' visually hidden.
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.yaml:5]: Should it be `"Emergency: Exit this page"`?

I can't help read "Emergency Exit…" as this sign

Emergency exit sign

File: packages/govuk-frontend/src/govuk/components/exit-this-page/template.njk [hunk 3]
```diff
@@ -8,8 +13,8 @@
   {%- if params.pressOneMoreTimeText %} data-i18n.press-one-more-time="{{ params.pressOneMoreTimeText | escape }}"{% endif %}
 >
   {{- govukButton({
-    html: params.html,
-    text: params.text | default("Exit this page"),
+    html: params.html if hasSetCustomText else defaultHtml,
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/exit-this-page/template.njk:16]: Or keep it short and sweet here? Parentheses optional

```suggestion
    html: params.html if (params.html or params.text) else defaultHtml,
```

File: packages/govuk-frontend/src/govuk/components/exit-this-page/template.njk (hunk 1 above)
Comment (querkmachine) [reply packages/govuk-frontend/src/govuk/components/exit-this-page/template.njk:3]: I may be misremembering (I originally wrote this about a month ago!) but I seem to recall it not working as expected without the `!== undefined`. I can give it a re-test just to be sure.

Comment (querkmachine) [reply packages/govuk-frontend/src/govuk/components/exit-this-page/template.njk:3]: Seems like it works fine without it, actually. I might be remembering a previous iteration, then.

File: CHANGELOG.md [hunk 4]
```diff
@@ -287,6 +287,21 @@ You don't need to change anything if you're using the `govukPagination` Nunjucks
 
 This change was introduced in [pull request #3899: Update default `aria-label` in Pagination component](https://github.com/alphagov/govuk-frontend/pull/3899).
 
+#### Update the Exit this Page button's default text
+
+The default text of the Exit this Page button has been updated to indicate that the button is a safety tool and not a generic method of leaving the current page. If you are using the component's default label, you may wish to update it to the new value.
+
+```diff
```
Comment (colinrotherham) [inline CHANGELOG.md:294]: 😮

File: CHANGELOG.md [hunk 5]
```diff
@@ -287,6 +287,21 @@ You don't need to change anything if you're using the `govukPagination` Nunjucks
...
+<a href="..." role="button" draggable="false" class="govuk-button govuk-button--warning govuk-exit-this-page__button govuk-js-exit-this-page-button" data-module="govuk-button">
+-   Exit this page
```
Comment (colinrotherham) [inline CHANGELOG.md:296]: ❤️

File: CHANGELOG.md [hunk 6]
```diff
@@ -287,6 +287,21 @@ You don't need to change anything if you're using the `govukPagination` Nunjucks
...
++   <span class="govuk-visually-hidden">Emergency</span> Exit this page
```
Comment (colinrotherham) [inline CHANGELOG.md:297]: 🙌

**Pull request merged**